File: tmp_edit.py
from pathlib import Path
path = Path('lib/agent/conversation-state-service.ts')
text = path.read_text()
old = "export interface ConversationStateData {\n  phase: \"clarifying\" | \"proposing\" | \"executing\" | \"completed\";\n  entities: Record<string, any>;\n  workingMemory: {\n    partialIntent?: string;\n    lastTopic?: string;\n    accumulatedEntities?: Record<string, any>;\n  };\n  confidence: number;\n  clarificationCount: number;\n  agentSessionId?: string;\n}\n\n/**\n * Service for managing conversation state in the database\n * This ensures state persistence and security (no client control)\n */\nexport class ConversationStateService {"
new = "export interface PendingConfirmation {\n  planId: string;\n  description?: string;\n  summary?: string;\n  stepCount?: number;\n  planSnapshot?: any;\n}\n\nexport interface ConversationStateData {\n  phase: \"clarifying\" | \"proposing\" | \"executing\" | \"completed\";\n  entities: Record<string, any>;\n  workingMemory: {\n    partialIntent?: string;\n    lastTopic?: string;\n    accumulatedEntities?: Record<string, any>;\n  };\n  pendingConfirmation?: PendingConfirmation | null;\n  confidence: number;\n  clarificationCount: number;\n  agentSessionId?: string;\n}\n\n/**\n * Service for managing conversation state in the database\n * This ensures state persistence and security (no client control)\n */\nexport class ConversationStateService {"
if old not in text:
    raise SystemExit('Original block not found')
text = text.replace(old, new, 1)
path.write_text(text)

File: tmp_edit.js
const fs = require("fs");
const path = "lib/agent/conversation-state-service.ts";
let text = fs.readFileSync(path, "utf8");
const oldBlock = `export interface ConversationStateData {
  phase: "clarifying" | "proposing" | "executing" | "completed";
  entities: Record<string, any>;
  workingMemory: {
    partialIntent?: string;
    lastTopic?: string;
    accumulatedEntities?: Record<string, any>;
  };
  confidence: number;
  clarificationCount: number;
  agentSessionId?: string;
}

/**
 * Service for managing conversation state in the database
 * This ensures state persistence and security (no client control)
 */
export class ConversationStateService {`;
const newBlock = `export interface PendingConfirmation {
  planId: string;
  description?: string;
  summary?: string;
  stepCount?: number;
  planSnapshot?: any;
}

export interface ConversationStateData {
  phase: "clarifying" | "proposing" | "executing" | "completed";
  entities: Record<string, any>;
  workingMemory: {
    partialIntent?: string;
    lastTopic?: string;
    accumulatedEntities?: Record<string, any>;
  };
  pendingConfirmation?: PendingConfirmation | null;
  confidence: number;
  clarificationCount: number;
  agentSessionId?: string;
}

/**
 * Service for managing conversation state in the database
 * This ensures state persistence and security (no client control)
 */
export class ConversationStateService {`;
if (!text.includes(oldBlock)) {
  throw new Error("Original block not found");
}
text = text.replace(oldBlock, newBlock);
fs.writeFileSync(path, text, "utf8");
